### ISSUE TITLE
Is RaftKeeper just taken from HouseKeeper?

### ISSUE THREAD

**Issue by zhanglistar** [JDRaftKeeper/RaftKeeper#32]
The source code structure is really close, check https://github.com/ClickHouse/ClickHouse/tree/master/src/Coordination.

**Comment by marising** [comment]
The framework and ZooKeeper protocol are almost the same as ClickHouse Keeper, but the Log, Snapshot, StateMachine, batch execution, and support for hundreds of thousands of Client connections are developed by ourselves. Of course, the performance difference is relatively large.

**Comment by zhanglistar** [comment]
Do you have tested with Jespen?

**Comment by marising** [comment]
I remember that ClickHouse Keeper has been done before, because they all use the same nuraft framework, so we didn't do it alone, but we plan to do it recently.

**Comment by JackyWoo** [comment]
@zhanglistar Thanks for your conern about the RaftKeeper project. The flowing is the additional discription of @marising .

As discribed in project ReadME, RaftKeeper is drived from ClickHouse. Some core modules is originally from ClikHouse, such as dispatcher, connection handler, keeper store, acl map. 

At the beginning we plan to contribute to ClickHouse. After some contributions such as 4lw command, reconnection(ignored by ClickHouse community),  we found that the goals of two projects are not consistent. HouseKeeper mainly supports ClickHouse, while RaftKeeper is moving towards the direction of being widely compatible with Zookeeper scenarios, such as HBase. 

Therefore, we decided to develop the projects independently. We have done a lot of rewriting on the core modules mentioned above and added some differentiated features compared to HouseKeeper, such as NIO network framework, support for reconnecting in session timeout, a new session management module, support read-write-consistency in session and  a new execution pipeline. 

And now RaftKeepeer an HouseKeepeer is quite different bouth in functions and internal implementation.
